packages/askar/src/wallet/AskarBaseWallet.ts

```diff
@@ -339,18 +397,25 @@ export abstract class AskarBaseWallet implements Wallet {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     const recipientKids: string[] = protectedJson.recipients.map((r: any) => r.header.kid)
 
-    for (const recipientKid of recipientKids) {
-      const recipientKeyEntry = await this.session.fetchKey({ name: recipientKid })
-      try {
-        if (recipientKeyEntry) {
-          return didcommV1Unpack(messagePackage, recipientKeyEntry.key)
+    // TODO: how long should sessions last? Just for the duration of the unpack? Or should each item in the recipientKids get a separate session?
```

Comment:
I don't know what's exactly the overhead of getting references to sessions in the pool (probably not much/nearly zero) but in this case I think it will become clearer if a single session is used for the whole unpack process (especially once we switch to the `using` keyword).